### PR TITLE
Feature/performance upgrades

### PR DESCRIPTION
**Pull request by sjdines**

### CODE REVIEW

File: src/environments/tests.py [hunk 1]
```diff
@@ -34,4 +36,59 @@ def test_create_identity_should_assign_relevant_attributes(self):
...
+        identity_2 = Identity.objects.create(
+            identifier="test-identity-2",
+            environment=self.environment,
+        )
+        identity_3 = Identity.objects.create(
+            identifier="test-identity-3",
+            environment=environment_2,
+        )
+
+        # User unassigned - automatically should be created via `Feature` save method.
```
Comment (matthewelwell) [inline src/environments/tests.py:55]: @sjdines The two unassigned calls to get FeatureState.objects in this section seem unnecessary?

Comment (sjdines) [reply src/environments/tests.py:55]: @matthewelwell thanks for the review; I will get to this as soon as I can (probably in the next few days)

Comment (matthewelwell) [reply src/environments/tests.py:55]: Thanks @sjdines. Let me know if there's anything I can do to help.

**Comment by matthewelwell** [comment]
Thanks for this @sjdines. Sorry for the delay. I've completed the merge now and just running through some final tests in our staging environment. I should be able to merge this change tomorrow.

**Pull request merged**